code/modules/clothing/neck/_neck.dm

```diff
@@ -18,36 +18,126 @@
 			. += mutable_appearance('icons/effects/blood.dmi', "maskblood")
 
 /obj/item/clothing/neck/tie
-	name = "tie"
+	name = "clip-on tie"
 	desc = "A neosilk clip-on tie."
 	icon = 'icons/obj/clothing/neck.dmi'
-	icon_state = "bluetie"
+	icon_state = "detective"
 	inhand_icon_state = "" //no inhands
 	w_class = WEIGHT_CLASS_SMALL
 	custom_price = PAYCHECK_CREW
 
-/obj/item/clothing/neck/tie/blue
+/obj/item/clothing/neck/tie/greyscale
+	greyscale_colors = "#4d4e4e"
+	name = "slick tie"
+	desc = "A neosilk tie."
+	icon_state = "tie_greyscale_tied"
+	greyscale_config = /datum/greyscale_config/ties
+	greyscale_config_worn = /datum/greyscale_config/ties_worn
+	flags_1 = IS_PLAYER_COLORABLE_1
+	/// All ties start untied unless otherwise specified
+	var/is_tied = FALSE
+	/// How long it takes to tie the tie
+	var/tie_timer = 4 SECONDS
+
+/obj/item/clothing/neck/tie/greyscale/Initialize(mapload)
+	. = ..()
+	update_appearance(UPDATE_ICON)
+	register_context()
+
+/obj/item/clothing/neck/tie/greyscale/AltClick(mob/user)
+	. = ..()
+	to_chat(user, span_notice("You concentrate as you begin [is_tied ? "untying" : "tying"] [src]..."))
+	var/tie_timer_actual = tie_timer
+	// Mirrors give you a boost to your tying speed. I realize this stacks and I think that's hilarious.
+	for(var/obj/structure/mirror/reflection in view(2, user))
+		tie_timer_actual /= 1.25
+	// Heads of staff are experts at tying their ties.
+	if(user.mind?.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
+		tie_timer_actual /= 2
+	// Tie/Untie our tie
+	if(!do_after(user, tie_timer_actual))
+		to_chat(user, span_notice("Your fingers fumble away from [src] as your concentration breaks."))
+		return
+	// Clumsy & Dumb people have trouble tying their ties.
+	if((HAS_TRAIT(user, TRAIT_CLUMSY) || HAS_TRAIT(user, TRAIT_DUMB)) && prob(50))
+		to_chat(user, span_notice("You just can't seem to get a proper grip on [src]!"))
+		return
+	// Success!
+	is_tied = !is_tied
+	user.visible_message("[user] adjusts [user.p_their()] tie[HAS_TRAIT(user, TRAIT_BALD) ? "." : " and runs a hand across [user.p_their()] head."]", span_notice("You successfully [is_tied ? "tied" : "untied"] [src]!"))
+	update_appearance(UPDATE_ICON)
+	user.update_clothing(ITEM_SLOT_NECK)
+
+/obj/item/clothing/neck/tie/greyscale/update_icon()
+	. = ..()
+	// Normal strip & equip delay, along with 2 second self equip since you need to squeeze your head through the hole.
+	if(is_tied)
+		icon_state = "tie_greyscale_tied"
+		strip_delay = 4 SECONDS
+		equip_delay_other = 4 SECONDS
+		equip_delay_self = 2 SECONDS
+	else // Extremely quick strip delay, it's practically a ribbon draped around your neck
+		icon_state = "tie_greyscale_untied"
+		strip_delay = 1 SECONDS
+		equip_delay_other = 1 SECONDS
+		equip_delay_self = 0
+
+/obj/item/clothing/neck/tie/greyscale/add_context(atom/source, list/context, obj/item/held_item, mob/user)
```

Comment:
The context is cool, maybe also put the hotkey into `examine()` for those who have it disabled?

Reply:
done